test/testdata/configs/gocritic.yml

```diff
@@ -2,9 +2,9 @@ linters-settings:
   gocritic:
     enabled-checks:
       - rangeValCopy
-      - flagDeref
```

Comment:
Else we received warning in stderr about enabling already enabled checks

Reply:
feel like a problem because this modifies the configuration and the expectation.

Reply:
under "problem" do you mean such kind of warnings?

I repeated the [existed logic](https://github.com/golangci/golangci-lint/pull/4335/files#diff-05b9162497599694184af35eedcc22cffa1179be8eff6935a6d082753349ba17L380):

```go
s.logger.Warnf("%s: no need to enable check %q: it's already enabled", goCriticName, enabledCheck)
```

but did not investigate why this is not working before 🤷‍♂️